packages/tools/devtools/devtools-view/src/DevtoolsView.tsx

```diff
@@ -170,22 +171,43 @@ export function DevtoolsView(): React.ReactElement {
 			handleIncomingMessage(message, inboundMessageHandlers, {
 				context: loggingContext,
 			});
+			clearTimeout(timeoutId); // Clear the timeout when cleaning up
 		}
 
 		messageRelay.on("message", messageHandler);
 
 		// Query for supported feature set
 		messageRelay.postMessage(getSupportedFeaturesMessage);
 
+		// Add a timeout for the query
+		const timeoutId = setTimeout(() => {
```

Comment:
Do we need to set this if we already have the `supportedFeatures`? I wonder if we should wrap this in a check to see if `supportedFeatures` is defined, and set `timeoutId` to undefined in that case?

Reply:
Hmm... So the situations will be like:
1. supportedFeatures is undefined/false, `supportedFeatures` is not found/supported, `timeout` is true meaning we are adding a timeout for query.
2. supportedFeatures is true, we found the supported features, `timeout` is false meaning we disable a timeout.

Reply:
The key thing is that there isn't a reason for us to be queuing the timeout in the first place if we already have our feature list from the webpage. I was picturing something like:

```typescript
let queryTimeout = undefined;
if (supportedFeatures === undefined) {
	queryTimeout = setTimeout(() => {
		...
	}
}
```

Reply:
Gotcha. Updated!